bee-bundle/src/transaction.rs

```diff
@@ -38,30 +36,59 @@ pub enum TransactionFieldError {
     FieldDeserializationError,
 }
 
-pub trait TransactionField: Sized {
-    fn try_from_tritbuf(buffer: TritBuf) -> Result<Self, TransactionFieldError>;
-    fn from_tritbuf_unchecked(buffer: TritBuf) -> Self;
+pub trait TransactionField: Sized + TransactionFieldType {
+    type inner;
+    fn try_from_inner(buffer: Self::inner) -> Result<Self, TransactionFieldError>;
+    fn from_inner_unchecked(buffer: Self::inner) -> Self;
 
-    fn into_inner(&self) -> TritBuf<T1B1Buf>;
+    fn into_inner(&self) -> &Self::inner;
+
+    fn trit_len() -> usize;
 }
 
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
-pub struct Payload(pub TritBuf<T1B1Buf>);
+pub trait  NumTritsOFValue {
+    fn num_trits(&self) -> usize;
+}
 
-impl Payload {
-    pub fn zeros() -> Self {
-        Self(TritBuf::zeros(PAYLOAD.trit_offset.length))
+
+pub trait TransactionFieldType {
+    type inner_type : NumTritsOFValue;
+
+    fn is_trits_type() -> bool;
+}
+
+
+impl NumTritsOFValue for TritBuf<T1B1Buf> {
+    fn num_trits(&self) -> usize {
+        self.len()
     }
+}
 
-    pub fn from_str(payload: &str) -> Self {
-        assert!(payload.len() <= PAYLOAD.tryte_offset.length);
-        assert!(payload.chars().all(|c| c.is_tryte()));
+impl NumTritsOFValue for i64 {
+    fn num_trits(&self) -> usize {
+        unimplemented!();
+    }
+}
 
-        let missing_trytes = PAYLOAD.tryte_offset.length - payload.len();
-        let payload = payload.to_owned() + &iter::repeat(TRYTE_ZERO).take(missing_trytes).collect::<String>();
-        let tritbuf = trytes_to_trits_buf(&payload);
+impl NumTritsOFValue for u64 {
+    fn num_trits(&self) -> usize {
+        unimplemented!();
+    }
+}
 
-        Self(tritbuf)
+impl NumTritsOFValue for usize {
+    fn num_trits(&self) -> usize {
+        unimplemented!();
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Payload(TritBuf<T1B1Buf>);
+
+impl Payload {
+
+    pub fn zeros() -> Self {
+        Self(TritBuf::zeros(PAYLOAD.trit_offset.length))
     }
 
     pub fn as_bytes(&self) -> &[i8] {
```

Comment:
Do you think this is still needed consider we are introducing  `TransactionField` trait?

Reply:
unfortunately yes,because there fields like `Index`/`Value`/`Timestamp` which should also implement `TransactionField` `as_bytes` and `zeros` does not make sense

Reply:
Btw, why do these fields "should" implement the trait ?

Reply:
Because they are transaction's fields and `from_inner` + `to_inner` make sense for them